main.go

```diff
@@ -5,13 +5,27 @@ import (
 	"net/http"
 )
 
+type State struct {
+	MyHasher
+	MyKVS
+	Stats
```

Comment:
IMHO this is a bit abusing anonymous embedding :-) 
Anonymous embedding should be used carefully, and only when it's necessary (in order to promote the embedding struct to an interface implemented by the embedded one) or if it somewhat improves the API (for example when you embed an sync.Lock).

Reply:
To sum it up. the default should be to give names to fields unless you can prove anonymous embedding is the way to go.